### ISSUE TITLE
Lack of Events for State Changes in MCV2_MultiToken.sol

### ISSUE THREAD

**Issue by sirius651** [Steemhunt/mint.club-v2-contract#43]
https://github.com/Steemhunt/mint.club-v2-contract/blob/main/contracts/MCV2_MultiToken.sol#L56

```
function mintByBond(address to, uint256 amount) external onlyBond {
        totalSupply += amount;
        _mint(to, 0, amount, "");
    }

    /**
     * @dev Burns tokens by the bonding curve contract.
     * Users can simply send tokens to the token contract address for the same burning effect without changing the totalSupply.
     * @param account The address from which the tokens will be burned.
     * @param amount The amount of tokens to burn.
     */
    function burnByBond(address account, uint256 amount) external onlyBond {
        if (amount > totalSupply) revert MCV2_MultiToken__BurnAmountExceedsTotalSupply();
        if(!isApprovedForAll(account, bond)) revert MCV2_MultiToken__NotApproved(); // `msg.sender` is always be `_bond`

        unchecked {
            totalSupply -= amount;
        }

        _burn(account, 0, amount);
    }

```

Just consider emitting events in mintByBond and burnByBond for better off-chain tracking of these state-changing operations.

**Comment by sydneyitguy** [comment]
I think the `_mint` and `_burn` functions will emit a `Transfer` event from the token implementation already, so having another event emission is unnecessary?

**Comment by sydneyitguy** [comment]
Merged with #44
